### ISSUE TITLE
How to handle errors?

### ISSUE THREAD

**Issue by Stock84-dev** [dovahcrow/bitmex-rs#17]
How to handle specific errors? For example to handle server overload it could be something along those lines. 
```rust
 let response = rest_client.request(post_order_request(bitmex_order)).await;
    match response {
        Ok(success) => {}, // do something with response
        Err(e) => {
            if e.message == "The system is currently overloaded. Please try again later." {
                thread::sleep(std::time::Duration::from_millis(500));
                // call again
            }
        },
    }
```
I can print out error:
`BitMEXErrorMessage { message: "The system is currently overloaded. Please try again later.", name: "HTTPError" }`
But don't know how to access its fields. Then tried to do downcast which gave me this error:
```rust
 if let Ok(err) = e.downcast::<bitmex::models::ErrorError>() {
                                    ^^^^^^^^ the trait `std::error::Error`
 is not implemented for bitmex::models::ErrorError`
```

**Comment by dovahcrow** [comment]
Hi, I just realized I forgot to reexport the error class. It is fixed in the new release v0.2.2. 

You can now just do 
```
if let Ok(err) = e.downcast::<bitmex::BitMEXError>() {
```
to get the underlying error.

**Comment by Stock84-dev** [comment]
Thanks, it's working now.
